### PR TITLE
W-15091391: make testing framework work out of the box from `4.6` onwards

### PR DESCRIPTION
**Pull request by rbourbonnavarro**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -827,6 +831,51 @@
...
+                </plugins>
+            </build>
+            <dependencies>
+                <dependency>
+                    <groupId>xml-apis</groupId>
+                    <artifactId>xml-apis</artifactId>
+                    <version>${xmlApisVersion}</version>
+                    <!-- This is needed only for jdk8, the required classes from here -->
+                    <!-- are already available in the jdk on versions 11+ -->
+                    <scope>provided</scope>
```
Comment (asanguinetti) [inline pom.xml:875]: shouldn't this be:
```suggestion
                    <scope>test</scope>
```
?

File: pom.xml [hunk 2]
```diff
@@ -827,6 +831,51 @@
...
+                        <artifactId>maven-dependency-plugin</artifactId>
+                        <version>3.5.0</version>
+                        <executions>
+                            <execution>
+                                <goals>
+                                    <goal>properties</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
```
Comment (asanguinetti) [inline pom.xml:852]: What would be the purpose of this one? Does it deserve a comment?

File: pom.xml (hunk 1 above)
Comment (rbourbonnavarro) [reply pom.xml:875]: The `tests-runner` needs the dependency to be declared as `provided` in order to put it in the container class loader.

File: pom.xml (hunk 2 above)
Comment (rbourbonnavarro) [reply pom.xml:852]: This is needed to set the value of the project property `${org.mule.apache:xerces2-xsd11:jar:}`, needed for `maven-surefire-plugin`.

**Review by asanguinetti — APPROVED** [review]

**Pull request merged**